api/src/main/java/au/org/aodn/nrmn/restapi/service/StagedRowService.java

```diff
@@ -20,19 +22,15 @@
     @Autowired
     StagedJobRepository jobRepo;
 
-    public Validated<ErrorInput, Integer> update(Long jobId, List<StagedRow> newRows) {
-        return jobRepo.findById(jobId).map(job -> {
-            newRows.forEach(row -> row.setStagedJob(job));
-            val saved = rowRepo.saveAll(newRows);
-            return Validated.<ErrorInput, Integer>valid(saved.size());
-        }).orElseGet(() ->
-                Validated.invalid(new ErrorInput("Couldn't find job: " + jobId, "job")));
-
-    }
-
-    public Validated<ErrorInput, Integer> delete( List<StagedRow> toDeleteRows) {
-        rowRepo.deleteAll(toDeleteRows);
-        rowRepo.flush();
-        return Validated.<ErrorInput, Integer>valid(toDeleteRows.size());
+    @Transactional
+    public Validated<ErrorInput, Integer> save(Long jobId, List<Long> toDeleteRowIds, List<StagedRow> newRows) {
+        rowRepo.deleteAllByIds(toDeleteRowIds);
+        Integer rowsAdded = 0;
```

Comment:
can't rows be updated as well?

Reply:
ATM the client only cares if rows are added so that it can perform a full reload in order to get the server-assigned row IDs.

Reply:
OK that sounds like it might result  in weird behaviour where you refresh and edits are gone but deletes and inserts saved.  For later then.